code/datums/gamemode/objectives/target/target.dm

```diff
@@ -46,15 +46,15 @@
 
 /datum/objective/target/proc/get_targets()
 	var/list/targets = list()
-	for(var/datum/mind/possible_target in ticker.minds)
-		if(possible_target != owner && ishuman(possible_target.current) && (possible_target.current.z != map.zCentcomm) && (possible_target.current.stat != DEAD) && !(possible_target.assigned_role in bad_assassinate_targets))
-			targets += possible_target
+	for(var/mob/living/carbon/human/H in mob_list)
+		if(H.mind && (H.mind.key != owner.key) && (H.z != map.zCentcomm) && !H.isDead() && !(H.mind.assigned_role in bad_assassinate_targets))
```

Comment:
Objectively slower than the mind list look-up.

Reply:
Speed isn't the point